cobaya/model.py

```diff
@@ -589,7 +589,7 @@ def logpost(self,
         return self.logposterior(params_values, make_finite=make_finite,
                                  return_derived=False, cached=cached).logpost
 
-    def get_valid_point(self, max_tries: int, ignore_fixed_ref: bool = False,
+    def get_valid_point(self, max_tries: Union[int, str], ignore_fixed_ref: bool = False,
```

Comment:
Hello @cmbant, thanks for the feedback on this! 

Indeed I think that, technically, that could be a float, as it is used just for evaluations like `if tries<max_tries`. Still, I wouldn't feel comfortable with having a floating `max_tries`, right? 

Also, that number comes from the `value` of a `NumberWithUnits`, already parsed, so that should be either a float or an int. Thus, I would drop the `Union` with `str` (going back as it was before).